include/util/graph_loader.hpp

```diff
@@ -62,32 +53,21 @@ inline unsigned loadRestrictionsFromFile(std::istream &input_stream,
  *  - list of traffic lights
  *  - nodes indexed by their internal (non-osm) id
  */
-inline NodeID loadNodesFromFile(std::istream &input_stream,
+inline NodeID loadNodesFromFile(std::string &filename,
                                 std::vector<NodeID> &barrier_node_list,
                                 std::vector<NodeID> &traffic_light_node_list,
                                 std::vector<extractor::QueryNode> &node_array)
 {
-    const FingerPrint fingerprint_valid = FingerPrint::GetValid();
-    FingerPrint fingerprint_loaded;
-    input_stream.read(reinterpret_cast<char *>(&fingerprint_loaded), sizeof(FingerPrint));
+    storage::io::FileReader file(filename, storage::io::FileReader::VerifyFingerprint);
+    NodeID number_of_nodes = file.ReadElementCount32();
+    SimpleLogger().Write() << "Importing number_of_nodes new = " << number_of_nodes << " nodes ";
 
-    if (!fingerprint_loaded.TestContractor(fingerprint_valid))
-    {
-        SimpleLogger().Write(logWARNING) << ".osrm was prepared with different build.\n"
-                                            "Reprocess to get rid of this warning.";
-    }
-
-    NodeID n;
-    input_stream.read(reinterpret_cast<char *>(&n), sizeof(NodeID));
-    SimpleLogger().Write() << "Importing n = " << n << " nodes ";
-
-    node_array.reserve(n);
+    node_array.reserve(number_of_nodes);
 
     extractor::ExternalMemoryNode current_node;
-    for (NodeID i = 0; i < n; ++i)
+    for (NodeID i = 0; i < number_of_nodes; ++i)
```

Comment:
Okay! So! After discussion with Karen and DanPat, I realize that we can't actually simply read directly from the disk data (from the file) into RAM as is, because the disk data structures is a vector of `ExternalMemoryNode`s which holds 5 pieces of information (lon, lat, id, barrier, traffic_light), while the RAM data structure is a vector of `QueryNode`s which contains 3 pieces of information (lon, lat, id). So we can't do a direct `ReadInto(data[i]);`. I've adopted it as far as I could though... :)